### DISCUSSION TITLE
Using Cache in Module

### DISCUSSION THREAD

**Discussion by overstartup** [OrchardCMS/OrchardCore#9973] [Q&A]
I have a Module as an API to retrieve data from my mobile application.
I want to use caching to retrieve data faster than calling from database, how I can add this functionality? There is any example to use?

**Comment by Skrypt** [comment]
Here an example of a service that uses caching (scalable) : 

https://github.com/OrchardCMS/OrchardCore/blob/d70bed827520689e1db29b1113d0fb3c4792c5f7/src/OrchardCore.Modules/OrchardCore.AdminMenu/Services/AdminMenuService.cs#L18-L23

Here is an other different example using a file cache : 

https://github.com/OrchardCMS/OrchardCore/blob/d70bed827520689e1db29b1113d0fb3c4792c5f7/src/OrchardCore.Modules/OrchardCore.Sitemaps/Cache/DefaultSitemapCacheProvider.cs#L19

Always depends on what you need exactly.
